### DISCUSSION TITLE
Focus

### DISCUSSION THREAD

**Discussion by Aytharn** [fsprojects/Avalonia.FuncUI#377] [General]
I am facing one very common task, having to focus the first element of a formulary when creating as new DB entry.
But I see no practical way with FuncUI... any help?

**Comment by JaggerJo** [comment]
I need more context to help.

Please provide a runnable example of the issue your facing, then I can help.

**Comment by Aytharn** [comment]
![Screenshot 2023-12-07 at 14 06 36](https://github.com/fsprojects/Avalonia.FuncUI/assets/8307825/3f82f154-58b4-4df2-a8fa-faf1f663bb9b)
This way should be more clear...
Then I woul be using this component this way
`let needFocus = state.Current.Mode = Add`
`FieldText.view settings (grid 0 0) (getLabel "first name") patient.Firstname fieldMode Required **needFocus** handlingFunc

**Reply by JaggerJo** [reply]
Please provide me with a small sample app and I'll take a look. 

It definitely works, but you need to use a different API + approach.

**Reply by Aytharn** [reply]
It does work, but it crashes internally once the element is disposed.
As far as I understand this is deeply broken inside Avalonia itself, TabNavigation is a dead feature, it's clear

**Reply by JaggerJo** [reply]
Mate, provide a repro and we'll find a way.

**Comment by Aytharn** [comment]
It will be quite some work, lemme try after a cup of tea…
